lxd/storage/backend_lxd.go

```diff
@@ -2223,6 +2225,225 @@ func (b *lxdBackend) CreateInstanceFromMigration(inst instance.Instance, conn io
+	}
+
+	isRemoteClusterMove := args.ClusterMoveSourceName != "" && b.driver.Info().Remote
+	if isRemoteClusterMove {
+		return fmt.Errorf("Conversion cannot be used for moving instances between members")
+	}
+
+	contentType := InstanceContentType(inst)
+	volType, err := InstanceTypeToVolumeType(inst.Type())
+	if err != nil {
+		return err
+	}
+
+	volStorageName := project.Instance(inst.Project().Name, inst.Name())
+	volConfig := make(map[string]string)
+	vol := b.GetNewVolume(volType, contentType, volStorageName, volConfig)
+
+	// Ensure storage volume settings are honored when doing migration.
+	vol.SetHasSource(false)
+	err = b.driver.FillVolumeConfig(vol)
+	if err != nil {
+		return fmt.Errorf("Failed filling volume config: %w", err)
+	}
+
+	// Check if the volume exists in database
+	dbVol, err := VolumeDBGet(b, inst.Project().Name, inst.Name(), volType)
+	if err != nil && !response.IsNotFoundError(err) {
+		return err
+	}
+
+	if dbVol != nil {
+		return fmt.Errorf("Volume for instance %q already exists in database", inst.Name())
+	}
+
+	// Check if the volume exists on storage.
+	volExists, err := b.driver.HasVolume(vol)
+	if err != nil {
+		return err
+	}
+
+	if volExists {
+		return fmt.Errorf("Volume already exists on storage but not in database")
+	}
+
+	revert := revert.New()
+	defer revert.Fail()
+
+	// Validate config and create database entry for new storage volume if not refreshing.
+	// Strip unsupported config keys (in case the export was made from a different type of storage pool).
+	err = VolumeDBCreate(b, inst.Project().Name, inst.Name(), args.Description, volType, false, vol.Config(), inst.CreationDate(), time.Time{}, contentType, false, true)
+	if err != nil {
+		return err
+	}
+
+	revert.Add(func() { _ = VolumeDBDelete(b, inst.Project().Name, inst.Name(), volType) })
+
+	// Generate the effective root device volume for instance.
+	err = b.applyInstanceRootDiskOverrides(inst, &vol)
+	if err != nil {
+		return err
+	}
+
+	// Override args.Name and args.Config to ensure volume is created based on instance.
+	args.Config = vol.Config()
+	args.Name = inst.Name()
+
+	if len(args.ConversionOptions) == 0 {
+		// If migration header supplies a volume size, then use that as block volume size instead of pool default.
+		// This way if the volume being received is larger than the pool default size, the block volume created
+		// will still be able to accommodate it.
+		if args.VolumeSize > 0 && contentType == drivers.ContentTypeBlock {
+			l.Debug("Setting volume size from offer header", logger.Ctx{"size": args.VolumeSize})
+			args.Config["size"] = fmt.Sprintf("%d", args.VolumeSize)
+		} else if args.Config["size"] != "" {
+			l.Debug("Using volume size from root disk config", logger.Ctx{"size": args.Config["size"]})
+		}
+
+		volCopy := drivers.NewVolumeCopy(vol)
+
+		err = b.driver.CreateVolumeFromMigration(volCopy, conn, args, nil, op)
+		if err != nil {
+			return err
+		}
+	} else {
+		var imgUploadPath string
```

Comment:
Do we need this var, would it be simpler & clearer to directly assign the intermediate location to `args.ConversionFilePath` in the if block for `format`, and leave it as default empty if not using `format` mode?